### ISSUE TITLE
Saving and loading.

### ISSUE THREAD

**Issue by JuiceGrape** [JuiceGrape/Incarnation-Of-Ragnarok#25]
Blocked by: #7 

As a player, I want my data to be saved so I can continue playing from where I left off when I start the game again.

To do:
- [ ] Pick a save location (make it modular, probably exe location or in documents to start with)
- [ ] Choose a json or binary approach for saving files (I prefer json as long as we don't intend to add multiplayer)
- [ ] Add a function that can save all items in the storage to a single file.
- [ ] Add a function that can load all items to the storage from a single file.
- [ ] Add a function that can save equipped gear to a single file.
- [ ] Add a function that can load equipped gear from a single file
- [ ] When loading a file, if the version in the file is mismatched from the game version, refuse loading.
- [ ] Make sure we can save to multiple profiles, but use a profile called "default" for everything for now.
- [ ] Automatically load the "default" profile on startup.

Notes:
- All save files will be prefixed by a line containing the version of the game.
- Don't allow saving during runs: Only in the hub area. (probably just an object that is only present in the hub area)

Out of scope:
- A menu to pick a loading option
- Auto saving: make it manual when pressing f12 for now.
